subsys/bluetooth/services/cts_c.c

```diff
@@ -0,0 +1,251 @@
+ *
+ * @retval 0        If the time struct is valid.
+ * @retval -EINVAL 	If the length does not match the expected size of the data.
+ */
+static int current_time_decode(current_time_char_t *p_time,
+			       uint8_t const *p_data, uint32_t const length)
+{
+	if (length != CTS_C_CURRENT_TIME_EXPECTED_LENGTH) {
+		/* Return to prevent accessing the out-of-bounds data. */
+		return -EINVAL;
+	}
+
+	uint32_t index = 0;
+
+	/* Date. */
+	p_time->exact_time_256.year = sys_get_le16(p_data + index);
+	index += sizeof(uint16_t);
+	p_time->exact_time_256.month = p_data[index++];
+	p_time->exact_time_256.day = p_data[index++];
+	p_time->exact_time_256.hours = p_data[index++];
+	p_time->exact_time_256.minutes = p_data[index++];
+	p_time->exact_time_256.seconds = p_data[index++];
+
+	/* Day of week. */
+	p_time->exact_time_256.day_of_week = p_data[index++];
+
+	/* Fractions of a second. */
+	p_time->exact_time_256.fractions256 = p_data[index++];
+
+	/* Reason for updating the time. */
+	p_time->adjust_reason.manual_time_update = (p_data[index] >> 0) & 0x01;
+	p_time->adjust_reason.external_reference_time_update =
+		(p_data[index] >> 1) & 0x01;
+	p_time->adjust_reason.change_of_time_zone = (p_data[index] >> 2) & 0x01;
+	p_time->adjust_reason.change_of_daylight_savings_time =
+		(p_data[index] >> 3) & 0x01;
+
+	return 0;
+}
+
+/**@brief Function for sanity check of a Current Time structure.
+ *
+ * @param[in] p_time  Current Time structure.
+ *
+ * @retval 0        If the time struct is valid.
+ * @retval -EINVAL  If the time is out of bounds.
+ */
+static int current_time_validate(current_time_char_t *p_time)
+{
+	/* Year. */
+	if ((p_time->exact_time_256.year > CTS_YEAR_MAX) ||
+	    ((p_time->exact_time_256.year < CTS_YEAR_MIN) &&
+	     (p_time->exact_time_256.year != 0))) {
+		return -EINVAL;
+	}
+
+	/* Month. */
+	if (p_time->exact_time_256.month > 12) {
+		return -EINVAL;
+	}
+
+	/* Day. */
+	if (p_time->exact_time_256.day > 31) {
+		return -EINVAL;
+	}
+
+	/* Hours. */
+	if (p_time->exact_time_256.hours > 23) {
+		return -EINVAL;
+	}
+
+	/* Minutes. */
+	if (p_time->exact_time_256.minutes > 59) {
+		return -EINVAL;
+	}
+
+	/* Seconds. */
+	if (p_time->exact_time_256.seconds > 59) {
+		return -EINVAL;
+	}
+
+	/* Day of week. */
+	if (p_time->exact_time_256.day_of_week > 7) {
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+/**@brief Function for handling of Current Time read response.
+ *
+ * @param[in] conn    Connection object.
+ * @param[in] err     ATT error code.
+ * @param[in] params  Read parameters used.
+ * @param[in] data    Pointer to the data buffer.
+ * @param[in] length  The size of the received data.
+ *
+ * @retval BT_GATT_ITER_STOP  Stop reading
+ */
+static uint8_t bt_gatt_cts_c_read_callback(struct bt_conn *conn, uint8_t err,
+					   struct bt_gatt_read_params *params,
+					   const void *data, uint16_t length)
+{
+	struct bt_gatt_cts_c *cts_c;
+	bt_gatt_cts_c_evt_t evt;
+	int err_code;
```

Comment:
Can be `err` ?

Reply:
The err variable is already used as ATT error code.